### PR TITLE
implement Baratheon cards from Someone Always Tells chapter pack

### PR DESCRIPTION
**Pull request by zacchiro**

### CODE REVIEW

**Comment by zacchiro** [comment]
french spoilers:
- https://i62.servimg.com/u/f62/17/07/69/48/27973710.jpg
- https://i.redditmedia.com/cU09a6qa6PX7_zqrGYg9UveeUAeZ8Q2FIiBdLfoMHF4.jpg?w=509&s=b05eb32a0caf23687b4c9249d1ed053c
english card names are guesswork of mine for now, so this should not be merged yet

File: server/game/cards/08.6-SAT/TheKingOnTheWall.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+const DrawCard = require('../../drawcard.js');
+
+class TheKingOnTheWall extends DrawCard {
+    setupCardAbilities(ability) {
+        this.attachmentRestriction({ faction: 'targaryen', unique: true });
```
Comment (DukeTax) [inline server/game/cards/08.6-SAT/TheKingOnTheWall.js]: Needs to be Baratheon.

File: server/game/cards/08.6-SAT/TheKingOnTheWall.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+            when: {
+                afterChallenge: event =>
+                    event.challenge.winner === this.controller &&
+                    event.challenge.challengeType === 'power' &&
+                    event.challenge.isParticipating(this.parent) &&
+                    this.numberOfOutOfFactionLocations() > 0 &&
+                    this.controller.canDraw()
+            },
+            handler: () => {
+                var cards = this.numberOfOutOfFactionLocations();
```
Comment (DukeTax) [inline server/game/cards/08.6-SAT/TheKingOnTheWall.js]: var --> let

File: server/game/cards/08.6-SAT/TheKingOnTheWall.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+                cards = this.controller.drawCardsToHand(cards).length;
+
+                this.game.addMessage('{0} uses {1} to draw {2} {3}',
+                    this.controller, this, cards, cards > 1 ? 'cards' : 'card');
+            }
+        });
+    }
+
+    numberOfOutOfFactionLocations() {
```
Comment (DukeTax) [inline server/game/cards/08.6-SAT/TheKingOnTheWall.js]: The contents of this method can be shortened to:

```javascript
return this.controller.getNumberOfCardsInPlay(card => card.getType() === 'location' && card.isOutOfFaction());
```

(It also makes sure neutral cards aren't counted as out of faction, as per the rules.)

File: server/game/cards/08.6-SAT/TheKingOnTheWall.js (hunk 2 above)
Comment (zacchiro) [reply server/game/cards/08.6-SAT/TheKingOnTheWall.js]: nope, as it is overwritten next line to retrieve the number of *actual* cards drawn

File: server/game/cards/08.6-SAT/TheKingOnTheWall.js (hunk 3 above)
Comment (zacchiro) [reply server/game/cards/08.6-SAT/TheKingOnTheWall.js]: thanks! I looked for something like this before my (bad) implementation, but couldn't find it :)

File: server/game/cards/08.6-SAT/TheKingOnTheWall.js (hunk 2 above)
Comment (DukeTax) [reply server/game/cards/08.6-SAT/TheKingOnTheWall.js]: It's still in the same block though so that should be fine? Not that it matters a lot.

Comment (zacchiro) [reply server/game/cards/08.6-SAT/TheKingOnTheWall.js]: fixed

**Pull request merged**